plugin/evm/atomic/vm/vm.go

```diff
@@ -0,0 +1,78 @@
+package vm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/codec"
+	avalanchedatabase "github.com/ava-labs/avalanchego/database"
+	avalanchecommon "github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/coreth/plugin/evm/atomic/sync"
+	"github.com/ava-labs/coreth/plugin/evm/message"
+
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+)
+
+var (
+	_ secp256k1fx.VM                     = &VM{}
+	_ block.ChainVM                      = &VM{}
+	_ block.BuildBlockWithContextChainVM = &VM{}
+	_ block.StateSyncableVM              = &VM{}
+)
+
+type ExtensibleEVM interface {
+	SetNetworkCodec(codec codec.Manager) error
+}
+
+type InnerVM interface {
+	ExtensibleEVM
+	avalanchecommon.VM
+	secp256k1fx.VM
+	block.ChainVM
+	block.BuildBlockWithContextChainVM
+	block.StateSyncableVM
+}
+
+type VM struct {
+	InnerVM // Inner EVM
+}
+
+func WrapVM(vm InnerVM) *VM {
+	return &VM{InnerVM: vm}
+}
+
+// Initialize implements the snowman.ChainVM interface
+func (vm *VM) Initialize(
```

Comment:
possibly instead of `plugin/evm/atomic/vm`, we can have `plugin/evm/coreth` which defines the VM for use in avalanchego.